packages/go_router/lib/src/route.dart

```diff
@@ -279,3 +322,64 @@ class GoRoute {
   ) =>
       const SizedBox.shrink();
 }
+
+/// A route that displays a UI shell around the matching child route. Builds
+/// a new Navigator that is used to display any matching sub-routes, instead
+/// of placing them on the root Navigator.
+///
+/// To display a child route on a different Navigator, provide it with a
+/// `navigatorKey` that matches the key provided to either the `GoRouter` or
+/// `ShellRoute` constructor.
+///
+/// The widget built by the matching child route becomes to the child parameter
+/// of the [builder].
+///
+/// For example:
+///
+/// ```
+/// ShellRoute(
+///   path: '/',
+///   builder: (BuildContext context, GoRouterState state, Widget child) {
+///     return Scaffold(
+///       appBar: AppBar(
+///         title: Text('App Shell')
+///       ),
+///       body: Center(
+///         child: child,
+///       ),
+///     );
+///   }
+/// ),
+/// ```
```

Comment:
Is this a valid code snippet? What is `child` if there are no child routes?

Reply:
Fixed in dbba9ba4f0663d2644719242d59c86c6ac66e153